recipes/root/all/conanfile.py

```diff
@@ -0,0 +1,305 @@
+import os
+import shutil
+import stat
+from glob import glob
+
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class PythonOption:
+    OFF = "off"
+    SYSTEM = "system"
+    # in future we may allow the user to specify a version when
+    # libPython is available in Conan Center Index.
+    ALL = [OFF, SYSTEM]
+    DEFAULT = OFF
+
+
+class RootConan(ConanFile):
+    name = "root"
+    version = "v6-22-02"
+    license = "LGPL-2.1-or-later"  # of ROOT itself, the recipe is under MIT license.
+    homepage = "https://root.cern/"
+    # ROOT itself is located at: https://github.com/root-project/root
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "CERN ROOT data analysis framework."
+    topics = ("data-analysis", "physics")
+    settings = ("os", "compiler", "build_type", "arch")
+    options = {
+        # Don't allow static build as it is not supported
+        # see: https://sft.its.cern.ch/jira/browse/ROOT-6446
+        # TODO: shared option should be reinstated when hooks issue is resolved
+        # (see: https://github.com/conan-io/hooks/issues/252)
+        # "shared": [True],
+        "fPIC": [True, False],
+        "python": PythonOption.ALL,
+    }
+    default_options = {
+        # "shared": True,
+        "fPIC": True,
+        # default python=off as there is currently no libpython in Conan center
+        "python": PythonOption.OFF,
+    }
+    generators = "cmake_find_package"
+    requires = (
+        "opengl/system",
+        "libxml2/2.9.10",
+        "glu/system",
+        "xorg/system",
+        "sqlite3/3.33.0",
+        "libjpeg/9d",
+        "libpng/1.6.37",
+        "libcurl/7.73.0",
+        "pcre/8.44",
+        "xz_utils/5.2.5",
+        "zstd/1.4.5",
+        "lz4/1.9.3",
+        "glew/2.1.0",
+        "openssl/1.1.1h",
+        "fftw/3.3.8",
+        "cfitsio/3.490",
+        "tbb/2020.3",
+        "libpng/1.6.37",
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(RootConan, self).__init__(*args, **kwargs)
+        self._cmake = None
```

Comment:
⚠️ not a python expert, this is how we usually do it

```suggestion

    _cmake = None
```

Reply:
My preference is to use an instance attribute, as opposed to a class attribute, in this case.

However, grepping the other conan-io/conan-center-index recipes I see that not a single one has an `__init__` method (🤯) so there is a clear overwhelming preference for class attributes over instance attributes.

I implemented this change for consistency with other recipes in  c4a2736.

Reply:
Thank you for showing me something new https://dzone.com/articles/python-class-attributes-vs-instance-attributes